Add error catching to locale.getlocale(locale.LC_ALL, "")

N-Hertstein opened the pull request:
Add error handling and fallbacks to en_US or C as language locale in case of misconfigured or minimal system.
Fixes issue #740

N-Hertstein committed:
Add error catching to locale.getlocale()

Add error handling and fallbacks to en_US or C as language locale in case of misconfigured or minimal system.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

N-Hertstein committed:
Change ' to " because test error...

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mizaki commented:
I don't know how possible it is but what if `C.UTF8` or `en_US.UTF-8` isn't available (`C` or `en_US` instead for example) etc.?

N-Hertstein committed:
Remove .UTF-8 from fallback language options

Modify fallback languages from en_US.UTF-8 & C.UTF-8 to en_US & C to avoid errors when UTF-8 is not available.

N-Hertstein commented:
> I don't know how possible it is but what if `C.UTF8` or `en_US.UTF-8` isn't available (`C` or `en_US` instead for example) etc.?

That would be better, fixed it.

mizaki commented:
What I mean is it's not known what will be on the system. So either the locales need to be interrogated and one picked or the best option might be to `pass` and log an exception.

lordwelch reviewed (changes requested):
Only use C as a fallback it will always exist and log that it's falling back to C `logger.error("Couldn't set the locale: unsupported locale setting; falling back to 'C' locale")`

N-Hertstein committed:
Fallback to C only and add Logging

Skip falling back to en_US and go straight to C as it is always available.
Add error logging.

N-Hertstein commented:
> Only use C as a fallback it will always exist and log that it's falling back to C `logger.error("Couldn't set the locale: unsupported locale setting; falling back to 'C' locale")`

Added as requested
